packages/validator/src/validator.ts

```diff
@@ -92,8 +95,25 @@ export class Validator {
       defaultFeeRecipient ?? defaultDefaultFeeRecipient
     );
     const indicesService = new IndicesService(logger, api, validatorStore, metrics);
-    const emitter = new ValidatorEventEmitter();
-    const chainHeaderTracker = new ChainHeaderTracker(logger, api, emitter);
+    // Do not enable doppelganger when started before genesis or first slot of genesis, because
+    // there would not have been any activity/signing in the network, so it is pointless to wait.
+    // Doppelganger should be enabled at slot > genesis slot, for example at slot 1, because
+    // same validator could have been started at slot 0 with signature already published to the network
+    if (opts.doppelgangerProtectionEnabled && getCurrentSlot(config, genesis.genesisTime) > 0) {
```

Comment:
I think this is not a good check:
in certain cases the doppelganger service just won't be started (even when the user specified `--doppelgangerProtectionEnabled`)

I'd recommend either conditionally calling `assertDoppelgangerSafe` or pass slot information into `assertDoppelgangerSafe` to conditionally do the check there.